packages/web/src/components/buy-sell-modal/BuySellModal.tsx

```diff
@@ -32,29 +38,101 @@ type TabOption = {
   const { isOpen, onClose } = useBuySellModal()
   const { spacing, color } = useTheme()
+  const { toast } = useContext(ToastContext)
   // const isMobile = useIsMobile() // Keep for potential mobile-specific adjustments - Removing for now
   const [activeTab, setActiveTab] = useState<BuySellTab>('buy')
   // selectedPairIndex will be used in future when multiple token pairs are supported
   const [selectedPairIndex] = useState(0)
 
+  // Transaction state
+  const [transactionData, setTransactionData] = useState<{
+    inputAmount: number
+    outputAmount: number
+    isValid: boolean
+  } | null>(null)
+
+  // Get the swap tokens mutation
+  const {
+    mutate: swapTokens,
+    status: swapStatus,
+    error: swapError
+  } = useSwapTokens()
+
   const tabs: TabOption[] = [
     { key: 'buy', text: messages.buy },
     { key: 'sell', text: messages.sell }
   ]
 
-  // Update token balances (placeholder - would connect to wallet)
-  useEffect(() => {
-    // This would be replaced with actual balance fetching logic
-    TOKENS.AUDIO.balance = 15000.0
-    TOKENS.USDC.balance = 100.0
-  }, [])
-
   const selectedPair = SUPPORTED_TOKEN_PAIRS[selectedPairIndex]
 
   const handleActiveTabChange = useCallback((newTab: BuySellTab) => {
     setActiveTab(newTab)
+    // Reset transaction data when changing tabs
+    setTransactionData(null)
   }, [])
 
+  const handleTransactionDataChange = useCallback(
+    (data: { inputAmount: number; outputAmount: number; isValid: boolean }) => {
+      setTransactionData(data)
+    },
+    []
+  )
+
+  // Handle continue button click
+  const handleContinueClick = useCallback(() => {
+    if (!transactionData || !transactionData.isValid) return
+
+    const { inputAmount } = transactionData
+
+    // Default slippage is 50 basis points (0.5%)
+    const slippageBps = 50
+
+    // Determine swap direction based on active tab
+    if (activeTab === 'buy') {
+      // Buy AUDIO with USDC
+      swapTokens({
+        inputMint: TOKEN_LISTING_MAP.USDC.address,
+        outputMint: TOKEN_LISTING_MAP.AUDIO.address,
+        amountUi: inputAmount,
+        slippageBps
+      })
+    } else {
+      // Sell AUDIO for USDC
+      swapTokens({
+        inputMint: TOKEN_LISTING_MAP.AUDIO.address,
+        outputMint: TOKEN_LISTING_MAP.USDC.address,
+        amountUi: inputAmount,
+        slippageBps
+      })
+    }
+  }, [activeTab, transactionData, swapTokens])
+
+  // Handle swap status changes
+  useEffect(() => {
+    if (swapStatus === 'success') {
+      toast(
+        activeTab === 'buy'
+          ? 'Successfully purchased AUDIO!'
+          : 'Successfully sold AUDIO!',
+        3000
+      )
+
+      // Close modal after 1 second on success
+      const timer = setTimeout(() => {
+        onClose()
+      }, 1000)
+
+      return () => clearTimeout(timer)
+    } else if (swapStatus === 'error') {
+      toast(swapError?.message || 'Transaction failed. Please try again.', 5000)
```

Comment:
generally we use messages object to share across mobile and web and transition to i18n eventually